src/libxdata/world/loaders/AirportLoader.cpp

```diff
@@ -108,16 +108,28 @@ void AirportLoader::onAirportLoaded(const AirportData& port) const {
             Location end1Loc(end1.latitude, end1.longitude);
             Location end2Loc(end2.latitude, end2.longitude);
             length = end1Loc.distanceTo(end2Loc) - end1.displace - end2.displace;
+        } else {
+            LOG_WARN("%s has runway with %d ends!", port.id.c_str(), entry.ends.size());
         }
 
-        for (auto &end: entry.ends) {
-            auto rwy = std::make_shared<Runway>(end.name);
-            rwy->setLocation(Location(end.latitude, end.longitude));
+        std::shared_ptr<Runway> end0 = NULL;

```

Comment:
`NULL` is deprecated, modern C++ has a new keyword for this: `nullptr`. This has some advantages, for example being of the type `nullptr_t`. That way, the compiler can do more optimizations, especially in template magic. Also, `shared_ptr` defaults to a nullpointer in construction, so you can simply remove the explicit initialization.